### PR TITLE
fix goToDefinition: refine contextValue and when-clause

### PR DESCRIPTION
**Pull request by Melody618**
this pr fixes the wrong entries of `goToDefinition` and refine context value and when-clause for future use.

format contextValue: `Dependency[+<type>]`
possible type: conflict, root

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -449,17 +449,17 @@
         },
         {
           "command": "maven.project.excludeDependency",
-          "when": "view == mavenProjects && viewItem =~ /^(Dependency)|(conflictDependency)$/",
+          "when": "view == mavenProjects && viewItem =~ /^(?!.*?\\+root)/",
```
Comment (Eskibear) [inline package.json:452]: will the command show on other non-dependency node?

**Review by Eskibear — CHANGES_REQUESTED** [review]
let's follow the same convention with https://github.com/microsoft/vscode-java-dependency/wiki/Register-Command-onto-the-Nodes-of-Project-View 
Then you can use the same regexp for when clauses.

i.e. 
`maven:dependency+conflict`

File: package.json (hunk 1 above)
Comment (Eskibear) [reply package.json:452]: @jdneo 
Here we have these node types: `maven:dependency[+root/+conflict]`. Is there a regexp to query "all dependency nodes with no +root property"? (I assume there should be a negative lookahead mechanism)

@Melody618  If the answer is no, we can also workaround it, by adding `+nonRoot` to non-root nodes.

Comment (jdneo) [reply package.json:452]: Try `maven:dependency(?=.*?\b\+conflict\b)(?!.*?\b\+root\b)`

Comment (Eskibear) [reply package.json:452]: image

Cool.

@Melody618 please use above regexp.
@jdneo suggest to add `(?!.*?\b\+exclude_prop\b)` in Wiki.

Comment (Melody618) [reply package.json:452]: as the picture shows, the regexp is not able to hit `maven:dependency`

Comment (Eskibear) [reply package.json:452]: of course it's as expected. for your case, simply append `(?!.*?\b\+root\b)`.

**Review by Eskibear — APPROVED** [review]

**Pull request merged**